pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -170,3 +172,17 @@ func (p *InstanceTypeProvider) isDefaultInstanceType(instanceTypeInfo *packing.I
 		!*instanceTypeInfo.BareMetal &&
 		functional.HasAnyPrefix(*instanceTypeInfo.InstanceType, "m", "c", "r", "a", "t3", "t4")
 }
+
+func (p *InstanceTypeProvider) isArchitectureSupported(architecture *string, instance *packing.Instance) bool {
+	return architecture == nil ||
```

Comment:
Aren't we guaranteed that architecture has been defaulted?

Reply:
No, only because of `GetAllInstanceTypeNames` in InstanceTypeProvider. It's used to get all the instance types for validating the provisioner spec, so it needs no filtering, but I would prefer to use the same overall logic to retrieve.